app/Composers/Modules/ComponentsComposer.php

```diff
@@ -33,11 +53,28 @@ class ComponentsComposer
     public function compose(View $view)
     {
         // Component & Component Group lists.
-        $usedComponentGroups = Component::enabled()->where('group_id', '>', 0)->groupBy('group_id')->pluck('group_id');
-        $componentGroups = ComponentGroup::whereIn('id', $usedComponentGroups)->orderBy('order')->get();
-        $ungroupedComponents = Component::enabled()->where('group_id', 0)->orderBy('order')->orderBy('created_at')->get();
+        $groupedComponents = $this->getVisibleGroupedComponents();
+        $ungroupedComponents = Component::ungrouped()->get();
 
-        $view->withComponentGroups($componentGroups)
+        $view->withComponentGroups($groupedComponents)
             ->withUngroupedComponents($ungroupedComponents);
     }
+
+    /**
+     * Get visible grouped components.
+     *
+     * @return \Illuminate\Support\Collection
+     */
+    protected function getVisibleGroupedComponents()
+    {
+        $componentGroupsBuilder = ComponentGroup::visible();
+        if ($this->guard->check()) {
```

Comment:
It stops the ability of two builders being instantiated. So we're not just randomly wasting resources :grimacing: